rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java

```diff
@@ -432,6 +486,93 @@ private static native boolean nativeCheckGoalExists(
     long goalInfoFromJavaConverterHandle,
     long goalInfoDestructorHandle);
 
+  private boolean goalExists(action_msgs.msg.GoalInfo goalInfo) {
+    long goalInfoFromJavaConverterHandle = goalInfo.getFromJavaConverterInstance();
+    long goalInfoDestructorHandle = goalInfo.getDestructorInstance();
+
+    return nativeCheckGoalExists(
+      this.handle, goalInfo, goalInfoFromJavaConverterHandle, goalInfoDestructorHandle);
+  }
+
+  private static native void nativePublishStatus(long handle);
+
+  private void publishStatus() {
+    this.nativePublishStatus(this.handle);
+  }
+
+  private static native void nativePublishFeedbackMessage(
+    long handle,
+    FeedbackMessageDefinition feedbackMessage,
+    long feedbackMessageFromJavaConverterHandle,
+    long feedbackMessageDestructorHandle);
+
+  private void publishFeedbackMessage(FeedbackMessageDefinition<T> feedbackMessage) {
+    this.nativePublishFeedbackMessage(
+      this.handle,
+      feedbackMessage,
+      feedbackMessage.getFromJavaConverterInstance(),
+      feedbackMessage.getDestructorInstance());
+  }
+
+  private static native void nativeNotifyGoalDone(long handle);
+
+  private void notifyGoalDone() {
+    this.nativeNotifyGoalDone(this.handle);
+  }
+
+  private static native void nativeExpiredGoals(
+    long handle, action_msgs.msg.GoalInfo goalInfo,
+    long goalInfoToJavaConverterHandle, Consumer<action_msgs.msg.GoalInfo> onExpiredGoal);
+
+  private void expiredGoals() {
```

Comment:
nit: I would rename this method (and the native method) to be present tense, `expireGoals` (no "d"). It seems more natural to me, since calling this method is the act of expiring goals.

Reply:
Sounds good, fixed in https://github.com/osrf/ros2_java/pull/65/commits/e0cc5e5aa964f54a95883098e68e15f6cfd18fc2.